### PR TITLE
UX: hide the list of file extensions on the upload dialog

### PR DESCRIPTION
**Pull request by AndrewPrigorshnev**

### CODE REVIEW

File: app/assets/javascripts/discourse/app/components/details-under-link.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import Component from "@ember/component";
+
+export default Component.extend({
+  actions: {
```
Comment (jjaffeux) [inline app/assets/javascripts/discourse/app/components/details-under-link.js:4]: please attempt to use @action instead

File: app/assets/javascripts/discourse/app/controllers/upload-selector.js [hunk 2]
```diff
@@ -1,49 +1,51 @@
...
 import discourseComputed from "discourse-common/utils/decorators";
-import { equal } from "@ember/object/computed";
+import { alias, equal } from "@ember/object/computed";
 
 export default Controller.extend(ModalFunctionality, {
   imageUrl: null,
   local: equal("selection", "local"),
   remote: equal("selection", "remote"),
   selection: "local",
+  showFormatsHint: alias("allowAdditionalFormats"),
```
Comment (jjaffeux) [inline app/assets/javascripts/discourse/app/controllers/upload-selector.js:17]: are you sure you need alias here ?

File: app/assets/javascripts/discourse/app/templates/components/details-under-link.hbs [hunk 3]
```diff
@@ -0,0 +1,9 @@
+{{#unless showDetails}}
+  {{d-button action=(action "showDetails")
```
Comment (jjaffeux) [inline app/assets/javascripts/discourse/app/templates/components/details-under-link.hbs:2]: nitpicking here but as you are new to the code base, the correct indentation here is:

```
{{d-button
  action=(action "showDetails")
  class="btn-link"
  label=label
  aria-label=label
}}
```

**Comment by jjaffeux** [comment]
I'm slightly annoyed that we call it `details-under-link` and that actually it's not a link :p

Also I wonder about the accessibility of this?

File: app/assets/javascripts/discourse/app/controllers/upload-selector.js (hunk 2 above)
Comment (AndrewPrigorshnev) [reply app/assets/javascripts/discourse/app/controllers/upload-selector.js:17]: Wasn't sure about it :) I'd better remove it then.

**Comment by AndrewPrigorshnev** [comment]
> I'm slightly annoyed that we call it `details-under-link` and that actually it's not a link :p

Sure, let's change it then. I've chosen this name because this component is a kind of a "link" that turns into a description. It's not an expander, it turns into a description and there's is no way back.

And we use `d-button` with the class `btn-link` for such "links".

I understand that it could be confusing since it isn't a link to another page. Would be `details-under-button` or `details-under-button-link` better? Or do you have something completely different in mind?

https://user-images.githubusercontent.com/1274517/116121785-04df0600-a6d2-11eb-9a8a-facc799dfa99.mov

**Comment by AndrewPrigorshnev** [comment]
> Also I wonder about the accessibility of this?

I will add some accessibility support.

**Comment by jjaffeux** [comment]
> > I'm slightly annoyed that we call it `details-under-link` and that actually it's not a link :p
> 
> Sure, let's change it then. I've chosen this name because this component is a kind of a "link" that turns into a description. It's not an expander, it turns into a description and there's is no way back.
> 
> And we use `d-button` with the class `btn-link` for such "links".
> 
> I understand that it could be confusing since it isn't a link to another page. Would be `details-under-button` or `details-under-button-link` better? Or do you have something completely different in mind?
> 
>  Screen.Recording.2021-04-26.at.20.39.32.mov

I think a good name would be a name describing what it does and not the implementation

**Comment by AndrewPrigorshnev** [comment]
@jjaffeux, speaking of accessibility. The component is accessible by pressing <kbd>Tab</kbd> since it uses `d-button`. And I've added `aria-live="assertive"` so screenreaders will read details after expanding. To me, it looks like it's all that should be done here. Do you have any additional suggestions?

> I think a good name would be a name describing what it does and not the implementation.

What do you think about `hidden-details`? Maybe you can suggest another name?

**Comment by AndrewPrigorshnev** [comment]
I've renamed the component, need the approval to merge.

**Review by jjaffeux — APPROVED** [review]

**Pull request merged**